### PR TITLE
Add envoyproxy image

### PR DESCRIPTION
**Pull request by jmguzik**

### CODE REVIEW

File: clusters/app.ci/supplemental-ci-images/envoyproxy/envoyproxy.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  failedBuildsHistoryLimit: 2
+  output:
+    to:
+      kind: ImageStreamTag
+      name: envoyproxy:tp-1.25.3
+  source:
+    dockerfile: |
+      FROM quay.io/centos/centos:stream9
+      LABEL maintainer="jguzik@redhat.com"
+      RUN curl -L -o /bin/envoy https://github.com/envoyproxy/envoy/releases/download/v1.25.3/envoy-1.25.3-linux-x86_64 && chmod +x /bin/envoy
```
Comment (droslean) [inline clusters/app.ci/supplemental-ci-images/envoyproxy/envoyproxy.yaml:26]: How can we maintain this for different architectures?

File: clusters/app.ci/supplemental-ci-images/envoyproxy/envoyproxy.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+spec:
+  successfulBuildsHistoryLimit: 1
+  failedBuildsHistoryLimit: 2
+  output:
+    to:
+      kind: ImageStreamTag
+      name: envoyproxy:tp-1.25.3
+  source:
+    dockerfile: |
+      FROM quay.io/centos/centos:stream9
```
Comment (droslean) [inline clusters/app.ci/supplemental-ci-images/envoyproxy/envoyproxy.yaml:24]: We already have this image in our clusters.

File: clusters/app.ci/supplemental-ci-images/envoyproxy/envoyproxy.yaml (hunk 1 above)
Comment (jmguzik) [reply clusters/app.ci/supplemental-ci-images/envoyproxy/envoyproxy.yaml:26]: why we should? This is only deployment for app.ci

**Pull request merged**